transformer/app.js

```diff
@@ -101,21 +118,122 @@ function transformIssue(body, type, accountId) {
   });
 }
 
+const generateUpdatePayload = (existingFindingsToUpdate) => existingFindingsToUpdate.map((finding) => ({
+  FindingIdentifiers: [{
+    Id: finding.Id,
+    ProductArn: finding.ProductArn,
+  }],
+  Severity: {
+    Label: finding.FindingProviderFields.Severity.Label,
+  },
+}));
+
+const writeParams = (id, timestamp) => ({
+  TableName: 'xray-findings',
+  Item: {
+    ID: id,
+    TIMESTAMP: timestamp,
+  },
+});
+
+const writeFindingsToDB = async (findingsCollection) => {
+  const promises = [];
+  findingsCollection.map(async (finding) => {
+    const id = finding.Id;
+    const timestamp = new Date().getTime().toString();
+    try {
+      promises.push(ddbClient.send(new PutCommand(writeParams(id, timestamp))));
+      console.log(`Writing ID to DB: ${finding.Id}`);
+    } catch (err) {
+      console.error(err);
+    }
+  });
+  const results = await Promise.allSettled(promises);
+  const resultsFulfilled = results.filter((result) => (result.status === 'fulfilled')).map((result) => result.status);
+  const resultsFailed = results.filter((result) => (result.status !== 'fulfilled')).map((result) => result.message);
```

Comment:
The return value in the map function should be `result.reason`, right?

Reply:
It's `message` in this case. No `reason` in the promise body for this call.

Reply:
Are you 100% sure? The spec says `result` will be an object with `status` and `reason`/`value`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#examples